packages/block-library/src/navigation/style.scss

```diff
@@ -77,11 +77,18 @@ $navigation-height: 60px;
 
 .wp-block-navigation {
 
-	// Min height and vertical alignment.
+	// Min height and vertical alignment (main menu - level 0)
 	&,
 	> .wp-block-navigation__container {
-		min-height: $navigation-height;
+		min-height: $navigation-min-height;
 		align-items: center;
+
+		> .wp-block-navigation-link {
+			display: flex;
+			min-height: $navigation-min-height;
+			margin-top: 0;
+			margin-bottom: 0;
+		}
```

Comment:
@frontdevde, not sure if it's the correct implementation. The idea behind this is to fit the `.wp-block-navigation-link` in height to its container.

<img src="https://user-images.githubusercontent.com/77539/72985435-bb17b980-3dc4-11ea-91b3-6e6e6535a764.png" width="500px" />